File: Leetcode/2864. Maximum Odd Binary Number/solution.cpp
#include <string>
using namespace std;

class Solution {
public:
    string maximumOddBinaryNumber(string s) {
        int c = count(s.begin(), s.end(), '1');
        return string(c - 1, '1') + string(s.size() - c, '0') + '1';
    }
};

File: Leetcode/2864. Maximum Odd Binary Number/solution.rs
impl Solution {
    pub fn maximum_odd_binary_number(s: String) -> String {
        let count_ones = s.chars().filter(|&c| c == '1').count();
        let result = "1".repeat(count_ones - 1) + &"0".repeat(s.len() - count_ones) + "1";
        result
    }
}
